File: delightreading-server/src/main/java/com/delightreading/SpringApplicationContextUtil.java
package com.delightreading;

import org.springframework.beans.BeansException;
import org.springframework.context.ApplicationContext;
import org.springframework.context.ApplicationContextAware;

public class SpringApplicationContextUtil implements ApplicationContextAware {
    private static ApplicationContext applicationContext;

    @Override
    public void setApplicationContext(ApplicationContext appContext) throws BeansException {

        applicationContext = appContext;
    }

    public static <T> T lookUpBean(Class<T> tClass) {
        return applicationContext.getBean(tClass);
    }
}

File: delightreading-backend/src/main/kotlin/com/delightreading/SpringApplicationContextUtil.kt
package com.delightreading

import org.springframework.beans.BeansException
import org.springframework.context.ApplicationContext
import org.springframework.context.ApplicationContextAware
import org.springframework.stereotype.Service

@Service
class SpringApplicationContextUtil : ApplicationContextAware {

    companion object {
        var applicationContext: ApplicationContext? = null

        fun <T> lookUpBean(tClass: Class<T>): T {
            return applicationContext!!.getBean(tClass)
        }
    }

    @Throws(BeansException::class)
    override fun setApplicationContext(appContext: ApplicationContext) {
        applicationContext = appContext
    }

}